code/datums/components/electrified_buckle.dm

```diff
@@ -133,11 +134,32 @@
 	if(!istype(mob_to_buckle))
 		return FALSE
 
+	if (requested_overlays)
+		var/atom/movable/parent_as_movable = parent
+		parent_as_movable.cut_overlay(requested_overlays)
+		for (var/image/overlay_image in requested_overlays)
+			overlay_image.layer = ABOVE_MOB_LAYER
+		parent_as_movable.add_overlay(requested_overlays)
+
```

Comment:
i feel like this is something that should be handled on the overlay_update signal, what do u think?